lib/core.js

```diff
@@ -307,6 +307,15 @@ interface Iterable<T> {
     @@iterator(): Iterator<T>;
 }
 
+/* Generators */
+interface Generator<Y,R,N> {
+    @@iterator(): Iterator<Y>;
+    next(value?: N): IteratorResult<Y|R>;
+}
+
+// we use this signature when typing yield* expressions
+declare var $yieldDelegate: (<R,N>(g: Generator<*,R,N>, n:N) => R) & ((i: Iterable<*>, n:mixed) => void);
```

Comment:
The `void` return type might also be wrong. As in my example above, if the `Iterable` has a generator for an `Iterable`, that generator might return and that returned value will be used here.